### ISSUE TITLE
Image is not displayed on facebook, linkedin sharing

### ISSUE THREAD

**Issue by chaitanya-nyros** [huacnlee/social-share-button#131]
When I tried to pass the image to `social_share_button_tag` It wil not display the image on the facebook or in linkedin.

**Comment by huacnlee** [comment]
I don't know what is you say.

**Comment by huacnlee** [comment]
Which image on the Facebook?

**Comment by chaitanya-nyros** [comment]
I mean when I tried to share the image using `social_share_button_tag` like this `= social_share_button_tag(@category_to_display.display_name(I18n.locale),:url => "http://www.someurl.com/",:image => 'http://www.gettyimages.com/gi-resources/images/Embed/new/embed2.jpg', desc: "The summary of page", via: "MyTwitterName")` the sharing window of facebook and linkedin displys only the title and descrtption but the image is not displayed.

**Comment by huacnlee** [comment]
No, you can't present an image URL for Facebook share, because it will get by share url to find an image URL.

**Comment by chaitanya-nyros** [comment]
Ohh really.... thank you so much you save my time :+1:

**Comment by chaitanya-nyros** [comment]
@huacnlee is it possible to share the required image using this gem?

**Comment by huacnlee** [comment]
Tryout on this example page

https://ruby-china.org/posts/faster-rails--how-to-check-if-a-record-exists

```html
<div class="social-share-button" data-title="Faster Rails: How to Check if a Record Exists" data-img="" data-url="" data-desc="" data-via="">
<a rel="nofollow " data-site="twitter" class="ssb-icon ssb-twitter" onclick="return SocialShareButton.share(this);" title="分享到 Twitter" href="#"></a>
<a rel="nofollow " data-site="wechat" class="ssb-icon ssb-wechat" onclick="return SocialShareButton.share(this);" title="分享到 微信" data-wechat-footer="打开微信，点击底部的 “发现”，<br/> 使用 “扫一扫” 即可将网页分享至朋友圈。" href="#"></a>
<a rel="nofollow " data-site="facebook" class="ssb-icon ssb-facebook" onclick="return SocialShareButton.share(this);" title="分享到 Facebook" href="#"></a>
<a rel="nofollow " data-site="google_plus" class="ssb-icon ssb-google_plus" onclick="return SocialShareButton.share(this);" title="分享到 Google+" href="#"></a>
<a rel="nofollow " data-site="weibo" class="ssb-icon ssb-weibo" onclick="return SocialShareButton.share(this);" title="分享到 新浪微博" href="#"></a>
<a rel="nofollow " data-site="douban" class="ssb-icon ssb-douban" onclick="return SocialShareButton.share(this);" title="分享到 豆瓣" href="#"></a>
</div>
```

There is have no any image url setted on social-share-button, when you click "Facebook", Facebook will reads this page (via share URL) to find out an image (if exist).

![2017-03-21 11 54 37](https://cloud.githubusercontent.com/assets/5518/24132020/2e93a114-0e2d-11e7-9859-105f52eb3f3f.png)

**Comment by chaitanya-nyros** [comment]
Ok @huacnlee but this is also not work form me on facebook and linkedin but works as expected in google+

**Comment by procandi** [comment]
Hello, @chaitanya-nyros do you have upload application image on your facebook for developers?

**Comment by procandi** [comment]
I can show image after upload facebook application image.

**Comment by aruprakshit** [comment]
@procandi I am having same issue. What do you mean by "upload application image on your facebook" ?

**Comment by aruprakshit** [comment]
@huacnlee What if my images are stored in s3 and my app runs on Heroku?

**Comment by aruprakshit** [comment]
What I see [from code](https://github.com/huacnlee/social-share-button/blob/master/app/assets/javascripts/social-share-button.coffee#L39) is that, it never sends the image for facebook. Why we are not attaching the image?

**Comment by kevintanhongann** [comment]
The current Facebook sharer JS API needs an app id to be included in the query parameters in order to work. Need to log this as a separate issue.

**Comment by aruprakshit** [comment]
@kevintanhongann You mean the image issue is related to APP id?

**Comment by hugoh1995** [comment]
I'm having the same issue. 

The image on Facebook is not the url that I included in the social_share_button_tag but the default thumbnail that I've set through de developer console. 

It does not show the image either on Twitter.

It only works on Pinterest.

**Comment by connoro10** [comment]
I've still been unable to get it to work but I think the problem is being discussed well here 
http://stackoverflow.com/questions/6693499/how-can-i-set-a-website-image-that-will-show-as-preview-on-facebook

**Comment by rennyallen** [comment]
My image which I want to share is also not displayed on Facebook, and for guys who is having the same issue, just need to specify the URL parameter, like this:
<%= social_share_button_tag(@post.title, url: "https://secure.example.com/example.jpg") %>

When your page gets shared on Facebook, it will automatically look at the content of your page and choose images from your page that meet its share image criteria, you can learn more about this [https://stackoverflow.com/questions/1138460/how-does-facebook-sharer-select-images-and-other-metadata-when-sharing-my-url](here)

**Comment by Afoster2** [comment]
The image definitely doesn't display on Facebook. I used the tag you provided:
`= social_share_button_tag(@post.title, :url => post_url(@post), desc: @post.title, :image => "@post.image") `

And I've tried leaving `:image => ""` as well. No such luck.

Any advice here would be much apprecaited.

**Comment by Sohair63** [comment]
As per gem code: there is no image passed to sharer

```
SocialShareButton.openUrl("http://www.facebook.com/sharer/sharer.php?u=#{url}&display=popup&quote=#{desc}", 555, 400)
```

**Comment by decabeza** [comment]
Facebook automatically gets the image from [Open Graph Markup](https://developers.facebook.com/docs/sharing/webmasters?locale=en_US) tags. For example:

```erb
<meta id="ogimage" property="og:image" content="<%= image_url %>"/>
```

So, if you want to use `@post.image` you should include this tags inside your `<head>`, something like:

```erb
<meta id="ogimage" property="og:image" content="<%= @post.image %>"/>
```

Now when share the url Facebook show the image. 😉 

I recommend use [Facebook Sharing Debugger](https://developers.facebook.com/tools/debug/sharing/) to testing.

**Comment by Sohair63** [comment]
How about if gem should do this implicitly, by adding this meta tag to the header?

**Comment by Sohair63** [comment]
what is the content of your `social_share_button.rb` file? 
make sure you have `linkedin` in `allow_sites`

**Comment by amit-dhoundiyal** [comment]
SocialShareButton.configure do |config|
  config.allow_sites = %w(twitter facebook linkedin)
   #google_plus weibo qq douban google_bookmark delicious tumblr pinterest email wechat vkontakte xing reddit hacker_news telegram odnoklassniki
end
That's my social share rb file

**Comment by Sohair63** [comment]
may be can you show the screenshot for better understanding, 
I checked it with `version: 1.1.0` and its working fine.

**Comment by amit-dhoundiyal** [comment]
![linkedin](https://user-images.githubusercontent.com/40171429/47767596-6210c100-dcfa-11e8-9425-589a044ba1df.png)
That's the image of the linkedin  after login but in post i do not get the data.

**Comment by crowezie** [comment]
I'm having the same issue as @amit-dhoundiyal. When I click the LinkedIn social share button the pop up appears and I'm given the options to 'share in a post' or 'send as a private message'. When I click either option I'm just given a blank text field like the one @amit-dhoundiyal posted above for the post option. I've specified the title, url, and desc fields

My erb tag below works for Twitter, WeChat, email etc. but no text shows up for the LinkedIn window. 
`<%= social_share_button_tag("Book #{@profile.first_name} today.", url: new_profile_booking_url(@profile), desc: @profile.description) %>`

Should I open up a new issue?

**Comment by neeraj30** [comment]
> gg
